xinference/model/llm/pytorch/glm4v.py

```diff
@@ -56,19 +56,29 @@ def match(
             return True
         return False
 
-    def load(self, **kwargs):
+    def load(self):
         from transformers import AutoModelForCausalLM, AutoTokenizer
 
         device = self._pytorch_model_config.get("device", "auto")
         self._device = select_device(device)
-        self._device = "auto" if self._device == "cuda" else self._device
+
+        kwargs = {"device_map": self._device}
+        quantization = self.quantization
+        if quantization != "none":
+            if self._device == "cuda" and self._is_linux():
+                kwargs["device_map"] = "auto"
+                self._device = "auto"
+                if quantization == "4-bit":
+                    kwargs["load_in_4bit"] = True
+                elif quantization == "8-bit":
+                    kwargs["load_in_8bit"] = True
```

Comment:
In the else block, we failed to load the quantized model silently. Could you please add some logs to provide more information about the error?

Reply:
> In the else block, we failed to load the quantized model silently. Could you please add some logs to provide more information about the error?

![4v-4bit](https://github.com/xorbitsai/inference/assets/918617/363f0314-4d42-4b17-9117-1a06244ac954)
![4v-8bit](https://github.com/xorbitsai/inference/assets/918617/85abcfcc-01de-4359-aa10-0f7e9b4935db)

I can launch quantized modes successfully from web ui

Reply:
@Minamiyama What @codingl2k1 means is that If your environment is not `self._device == "cuda" and self._is_linux()`, when the users choose `4-bit` or `8-bit`, the model is still loaded via `none` quantization, which may lead to misunderstanding. You should add a log or just raise error to show this case.